src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -1645,7 +1645,7 @@ static void SR_handler(int sig, siginfo_t* siginfo, void* context) {
 
   // Save and restore errno to avoid confusing native code with EINTR
   // after sigsuspend.
-  int old_errno = errno;
+  ErrnoPreserver ep;
```

Comment:
I don't think this is incorrect but I will just note that by using the `ErrnoPreserver` we now restore `errno` on all the return paths below, not just for the "success" case.